packages/bpk-component-modal/src/BpkModalV2/BpkModal.tsx

```diff
@@ -112,32 +112,46 @@ export const BpkModalV2 = (props: Props) => {
   const ref = useRef<HTMLDialogElement>(null);
 
   useEffect(() => {
+    const dialog = document.getElementById(`${id}`);
+    const dialogWithPolyfill = document.getElementById(`${id}-polyfill`);
+
+    const handleBackdropClick = (modal: HTMLElement | null) => {
+      if (modal) {
+        modal.addEventListener('click', (event: Event) => {
+          const { target } = event;
+
+          if (target === modal) {
+            modal === dialog ? ref.current?.close?.() : onClose();
+          }
+        });
+      }
+    };
+
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (event.key === 'Escape') {
+        event.preventDefault();
+        onClose();
+      }
+    };
+
     if (isOpen) {
       ref.current?.showModal?.();
 
-      const dialog = document.getElementById(`${id}`);
-      const dialogWithPolyfill = document.getElementById(`${id}-polyfill`);
-
-      const handleBackdropClick = (modal: HTMLElement | null) => {
-        if (modal) {
-          modal.addEventListener('click', (event: Event) => {
-            const { target } = event;
-
-            if (target === modal) {
-              modal === dialog ? ref.current?.close?.() : onClose();
-            }
-          });
-        }
-      };
+      if (dialogWithPolyfill) {
+        handleBackdropClick(dialogWithPolyfill);
+        window.addEventListener('keydown', handleKeyDown);
+      }
```

Comment:
The polyfill needs to also cover the esc key as the dialog does not work on Safari <15.2